aws-xray-agent-plugin/build.gradle.kts

```diff
@@ -77,6 +77,42 @@ tasks {
         rename("disco-java-agent-$discoVer.jar", "disco-java-agent.jar")
     }
 
+    // JARs are just archives, so we can unzip it to a tmp folder
+    register<Copy>("unzipAgent") {
+        from(zipTree("$buildDir/libs/disco/disco-java-agent.jar"))
+        into("$buildDir/libs/tmp")
+
+        dependsOn("copyAgent")
+    }
+
+    // Inspect and rewrite the manifest in the tmp folder
+    register("rewriteManifest") {
+        val discoVer = rootProject.extra["discoVersion"]
+
+        // Reads the old manifest with versioned jar, then rewrites the manifest with replaced jar name
+        doFirst {
+            val oldManifest: List<String> = File("$buildDir/libs/tmp/META-INF/MANIFEST.MF").readLines()
+            File("$buildDir/libs/tmp/META-INF/MANIFEST.MF").printWriter().use { out ->
+                oldManifest.map {
+                    it.replace("disco-java-agent-$discoVer.jar", "disco-java-agent.jar")
+                }.forEach {
+                    out.println(it)
+                }
+            }
+        }
+
+        dependsOn("unzipAgent")
+    }
+
+    // Re-zip the tmp folder into the new agent JAR and overwrite the agent JAR with the old manifest
+    register<Zip>("rezipAgent") {
```

Comment:
Need to use `Jar` task I think or manifest won't be at top of archive (for compliance with jar format).

Reply:
I verified the META-INF/MANIFEST.MF is at the root of the new `disco-java-agent.jar` when I inspected it with `jar -tf` and the integ tests using it pass, is there something to else to JAR compliance I'm missing?

Reply:
Hmm - I'm surprised it'd be at the top given `from(...)` should just be going in alphabetical order. Perhaps newer versions of `jar` automatically deal with it, I'm not sure, but I've had issues in the past. Either way, Gradle has the `Jar` task, we should use it when creating jars to make sure it works OK. We could consider read, split lines / `:`, and use `Jar.manifest` instead of the printwriting too.